lib/partials.ts

```diff
@@ -96,12 +96,14 @@ export const findPartial = (
 			}),
 		(structureReferences: string[][]) =>
 			thru(structureReferences, (combinations) => {
-				const products = cartesianProductWith<string, string[]>(
-					combinations,
-					(accumulator: string[], element: string) =>
-						accumulator.concat([element]),
-					[[]],
-				);
+				const products = [
+					...cartesianProductWith<string, string[]>(
+						combinations,
+						(accumulator: string[], element: string) =>
+							accumulator.concat([element]),
```

Comment:
It should use less memory to do:
```suggestion
						(accumulator: string[], element: string) => {
							accumulator.push(element);
							return accumulator;
						}
```

Reply:
You are correct, though in practice, the iteratee is called generally on small arrays 

Also, in order to make the suggestion work, I need to change the `nextCartesianProduct` function and add a `clone`, which might offset the memory advantages of using push vs concat.

```
const newCombination = iteratee(clone(combination), value);
if (newCombination) {
	yield* nextCartesianProduct(sets, iteratee, newCombination, setRow + 1, 0);
}
``

Reply:
actually I should add the `clone` anyway, to protect the product calculation results from being messed with by the callback

Reply:
done